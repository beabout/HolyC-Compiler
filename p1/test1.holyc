int = 3+2
_dc = 3
"&!88"
'a'a
charptr hello = "use \n to denote a newline character" + "this"
'\n
"thing" # comment after thing
# aaaaaa # asdf
while (true) {
  _dc++;
}