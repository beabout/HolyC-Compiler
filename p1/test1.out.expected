INT [1,1]
ASSIGN [1,5]
INTLIT:3 [1,7]
CROSS [1,8]
INTLIT:2 [1,9]
ID:_dc [2,1]
ASSIGN [2,5]
INTLIT:3 [2,7]
STRINGLIT:"&!88" [3,1]
CHARLIT:a
CHARLIT:a
CHARPTR [5,1]
ID:hello [5,9]
ASSIGN [5,15]
STRINGLIT:"use \n to denote a newline character" [5,17]
CROSS [5,56]
STRINGLIT:"this" [5,58]
CHARLIT:newline
STRINGLIT:"thing" [7,1]
WHILE [9,1]
LPAREN [9,7]
TRUE [9,8]
RPAREN [9,12]
LCURLY [9,14]
ID:_dc [10,3]
CROSSCROSS [10,6]
SEMICOLON [10,8]
RCURLY [11,1]
EOF [11,2]
